kernel/src/sched.rs

```diff
@@ -20,10 +28,60 @@ use crate::process::{self, Task};
 use crate::returncode::ReturnCode;
 use crate::syscall::{ContextSwitchReason, Syscall};
 
-/// The time a process is permitted to run before being pre-empted
-const KERNEL_TICK_DURATION_US: u32 = 10000;
 /// Skip re-scheduling a process if its quanta is nearly exhausted
-const MIN_QUANTA_THRESHOLD_US: u32 = 500;
+pub(crate) const MIN_QUANTA_THRESHOLD_US: u32 = 500;
+
+/// Trait which any scheduler must implement.
+pub trait Scheduler<C: Chip> {
+    /// Return the next process to run, and the timeslice length for that process.
+    /// Passing None instead of an AppId will lead to nothing being run.
+    /// Passing None instead of a timeslice will cause the process to be run cooperatively
```

Comment:
```suggestion
    /// Return the next process to run, and the timeslice length for that process.
    ///
    /// The first argument is an optional `AppId` of the process to run next.
    /// Returning `None` instead of `Some(AppId)` means nothing will run. The second
    /// argument is an optional number of microseconds to use as the length of the process's
    /// timeslice. If an `AppId` is returned in the first argument, then returning `None`
    /// instead of a timeslice will cause the process
    /// to be run cooperatively (i.e. without preemption). Otherwise the process will run
    /// with a timeslice set to the specified length.
```